cmd/promtool/main.go

```diff
@@ -276,6 +276,9 @@ func main() {
 		p = &jsonPrinter{}
 	case "promql":
 		p = &promqlPrinter{}
+	case "unittest":
+		// TODO pick out default value used in query range (L892)
```

Comment:
Agree on the TODO and also it should actually be done before we can merge this as currently not specifying the `step` leads to zero interval in the output which then defaults to 1m which may or may not be what you want.

I think there's two options:
- move the calculation of the step (and start, end epoch time) out of `QueryRange` so we can reuse the step,
- or just say for MVP that we always write `1m` as the interval, regardless what step was used. That's a bit easier to understand for the user as they don't have to figure out what 0 means.

Reply:
Does the new approach fall a little closer in line with what you were thinking for the first option?